src/hooks/useOwner.ts

```diff
@@ -0,0 +1,46 @@
+//- by Alejo Viola and Leo
+
+import { useEffect, useState } from 'react';
+
+//- Web3 Imports
+import { useWeb3React } from '@web3-react/core';
+import { Web3Provider } from '@ethersproject/providers/lib/web3-provider';
+
+enum OwnerState {
+	Disconnected = 0,
+	Autorized = 1,
+	NoAuthorized = 2,
+}
+
+export const useOwner = () => {
+	//- Wallet Data
+	const walletContext = useWeb3React<Web3Provider>();
+	const { account } = walletContext;
+
+	const [owner, set] = useState('');
+
+	const setOwner = (address: string) => {
```

Comment:
Calling this method won't change the `isOwner` state variable since the effect below does not include `owner` in it's dependency list.